kommunicateui/src/main/java/com/applozic/mobicomkit/uiwidgets/conversation/richmessaging/adapters/KmVideoRMAdapter.java

```diff
@@ -0,0 +1,241 @@
+import android.widget.TextView;
+import android.widget.VideoView;
+
+import androidx.annotation.NonNull;
+import androidx.recyclerview.widget.RecyclerView;
+
+import com.applozic.mobicomkit.api.conversation.Message;
+import com.applozic.mobicomkit.uiwidgets.R;
+import com.applozic.mobicomkit.uiwidgets.conversation.richmessaging.callbacks.KmRichMessageListener;
+import com.applozic.mobicomkit.uiwidgets.conversation.richmessaging.models.KmRichMessageModel;
+import com.applozic.mobicomkit.uiwidgets.kommunicate.utils.KmThemeHelper;
+import com.applozic.mobicommons.json.GsonUtils;
+
+
+import java.util.Arrays;
+import java.util.List;
+
+public class KmVideoRMAdapter extends KmRichMessageAdapter {
+
+
+    private List<KmRichMessageModel.KmPayloadModel> payloadList;
+    double currentPos, totalDuration;
+
+
+    KmVideoRMAdapter(Context context, KmRichMessageModel model, KmRichMessageListener listener, Message message, KmThemeHelper themeHelper, boolean isMessageProcessed) {
+        super(context, model, listener, message, themeHelper);
+        if (model.getPayload() != null) {
+            this.payloadList = Arrays.asList((KmRichMessageModel.KmPayloadModel[])
+                    GsonUtils.getObjectFromJson(model.getPayload(), KmRichMessageModel.KmPayloadModel[].class));
+        }
+    }
+
+
+    @NonNull
+    @Override
+    public RecyclerView.ViewHolder onCreateViewHolder(@NonNull ViewGroup viewGroup, int i) {
+
+        View itemView = LayoutInflater.from(context).inflate(R.layout.km_rich_message_video_item, viewGroup, false);
+        return new RichMessageVideoHolder(itemView);
+
+
+    }
+
+    @Override
+    public void onBindViewHolder(@NonNull RecyclerView.ViewHolder viewHolder, int i) {
+        bindItems(viewHolder, i);
+    }
+
+    @Override
+    void bindItems(RecyclerView.ViewHolder viewHolder, int position) {
+        super.bindItems(viewHolder, position);
+        final RichMessageVideoHolder holder = (RichMessageVideoHolder) viewHolder;
+        if (payloadList != null) {
+            final KmRichMessageModel.KmPayloadModel payloadModel = payloadList.get(position);
+            WebSettings webSettings = holder.webview.getSettings();
+            webSettings.setJavaScriptEnabled(true);
+            holder.webview.setWebChromeClient(new WebChromeClient());
+
+            holder.webview.getSettings().setPluginState(WebSettings.PluginState.ON);
+            if (!TextUtils.isEmpty(payloadModel.getSource())) {
+                holder.videoViewRoot.setVisibility(View.GONE);
+                holder.captionText.setVisibility(View.GONE);
+                holder.webViewRoot.setVisibility(View.VISIBLE);
+                String currentUrl = "<iframe width=\"100%\" height=\"100%\" src=\"" + payloadModel.getUrl() + "\" frameborder=\"0\" allow=\"accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture\" allowfullscreen></iframe>";
+                holder.webview.loadData(currentUrl, "text/html", "utf-8");
```

Comment:
Will this always open the webview (any browser), any particualr reason for not opening the default app and webview?

Reply:
its not external webview. This webview xml component inside Recyclerview item layout.